### PR TITLE
Dynamically add robots to Simulator

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Users can now right-click in the simulator to bring up a small dialog with options to `Add Yellow Robot Here` and `Add Blue Robot Here`. This will allow robots to be added dynamically to the simulator like grsim.

### Testing Done

unit tests and manual testing

### Resolved Issues

#1536 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/simulation/threaded_simulator.cpp [hunk 1]
```diff
@@ -168,3 +168,13 @@ std::weak_ptr<PhysicsRobot> ThreadedSimulator::getRobotAtPosition(const Point &p
     std::scoped_lock lock(simulator_mutex);
     return simulator.getRobotAtPosition(position);
 }
+
+void ThreadedSimulator::addYellowRobot(const Point &position)
+{
+    simulator.addYellowRobot(position);
```
Comment (jonathanlew) [inline src/software/simulation/threaded_simulator.cpp]: do we need locking here?

**Review by kelvinkoon — DISMISSED** [review]
LGTM pending comment.

File: src/software/simulation/threaded_simulator.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/software/simulation/threaded_simulator.cpp]: Yup good catch. Fixed.

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by kelvinkoon — APPROVED** [review]

**Pull request merged**